src/jit/codegenarmarch.cpp

```diff
@@ -22,6 +22,91 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "gcinfo.h"
 #include "emit.h"
 
+//------------------------------------------------------------------------
+// genStackPointerConstantAdjustment: add a specified constant value to the stack pointer.
+// No probe is done.
+//
+// Arguments:
+//    spDelta                 - the value to add to SP. Must be negative or zero.
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genStackPointerConstantAdjustment(ssize_t spDelta)
+{
+    assert(spDelta < 0);
+
+    // We assert that the SP change is less than one page. If it's greater, you should have called a
+    // function that does a probe, which will in turn call this function.
+    assert((target_size_t)(-spDelta) <= compiler->eeGetPageSize());
```

Comment:
How does this assert not fire when it is called by `genStackPointerConstantAdjustmentLoopWithProbe`?

Reply:
The "loop with probe" function breaks up the probes into pageSize chunks, so a SP change is never more than one page when this is called.

Reply:
Ah yes, I see that described a bit further down. It's just a bit confusing because the function that's calling this one isn't actually the one that breaks it up.